app/services/settings-v2/video.ts

```diff
@@ -5,78 +5,230 @@ import { mutation, StatefulService } from '../core/stateful-service';
-      fpsType: context.fpsType,
-      fpsCom: `${context.fpsNum}-${context.fpsDen}`,
-      fpsNum: context.fpsNum,
-      fpsDen: context.fpsDen,
-      fpsInt: context.fpsNum,
+      width: context.video.baseWidth,
+      height: context.video.baseHeight,
     };
   }
 
+  get isGreen(): boolean {
+    return false;
+  }
+
   get videoSettings() {
-    return this.settingsManagerService.videoSettings;
+    return this.settingsManagerService.views.videoSettings;
   }
 
-  migrateSettings() {
-    Object.keys(this.videoSettings).forEach((key: keyof obs.IVideo) => {
-      this.SET_VIDEO_SETTING(key, this.videoSettings[key]);
-    });
+  getBaseResolution(display: TDisplayType = 'horizontal') {
+    return `${this.state[display].baseWidth}x${this.state[display].baseHeight}`;
+  }
+
+  migrateSettings(display?: TDisplayType) {
+    const contextName = contextNameMap[display];
 
-    if (invalidFps(this.state.videoContext.fpsNum, this.state.videoContext.fpsDen)) {
-      this.setVideoSetting('fpsNum', 30);
-      this.setVideoSetting('fpsDen', 1);
+    if (display === 'default' || display === 'horizontal') {
+      this.state.horizontal = this.state.horizontalContext.video;
+      Object.keys(this.state.horizontalContext.legacySettings).forEach(
+        (key: keyof obs.IAdvancedStreaming | keyof obs.ISimpleStreaming) => {
+          this.SET_VIDEO_SETTING(key, this.state.horizontalContext.legacySettings[key]);
+        },
+      );
+      Object.keys(this.state.horizontalContext.video).forEach((key: keyof obs.IVideo) => {
+        this.SET_VIDEO_SETTING(key, this.state.horizontalContext.video[key]);
+        if (
+          invalidFps(
+            this.state.horizontalContext.video.fpsNum,
+            this.state.horizontalContext.video.fpsDen,
+          )
+        ) {
+          this.createDefaultFps();
+        }
+      });
+    } else {
+      const data = this.videoSettings.green;
+
+      this.state[display] = this.state[contextName].video;
+      Object.keys(data).forEach(
+        (key: keyof obs.IAdvancedStreaming | keyof obs.ISimpleStreaming) => {
+          this.SET_VIDEO_SETTING(key, data[key], display);
+          if (
+            invalidFps(this.state[contextName].video.fpsNum, this.state[contextName].video.fpsDen)
+          ) {
+            this.createDefaultFps(display);
+          }
+        },
+      );
     }
   }
 
-  establishVideoContext() {
-    if (this.state.videoContext) return;
+  establishVideoContext(display: TDisplayType = 'horizontal') {
+    const contextName = contextNameMap[display];
+
+    if (this.state[contextName]) return;
 
-    this.SET_VIDEO_CONTEXT();
+    this.state[contextName] = obs.VideoFactory.create();
+    this.state[display] = {} as obs.IVideoInfo;
+    this.migrateSettings(display);
+    this.state[contextName].video = this.state[display];
 
-    this.migrateSettings();
-    obs.VideoFactory.videoContext = this.state.videoContext;
+    return !!this.state[contextName];
```

Comment:
For both `establishVideoContext` and `destroyVideoContext`, this was to check if the contexts were in fact created/destroyed to catch errors. This may not be the ideal way to do this so please let me know if you have any thoughts about implementing error handling for this.